apps/jupyter/extensions/notebook-runner/src/__tests__/task-exporter.spec.ts

```diff
@@ -0,0 +1,154 @@
+      taskTemplateFile: new Blob(["template"]),
+      studentTaskFile: new Blob(["student"]),
+      autograderFile: new Blob(["autograder"]),
+      data: new Map(),
+      gradingData: new Map(),
+      src: new Map(),
+      gradingSrc: new Map(),
+    };
+
+    const result = await TaskConverter.exportCrtInternalTask(task);
+
+    const resultZip = await loadExportedBlob(result);
+
+    const dataFiles = Object.keys(resultZip.files).filter((path) =>
+      path.startsWith(CrtInternalFiles.data),
+    );
+    expect(dataFiles).toHaveLength(0);
+  });
+});
+
+describe("exportExternalCustomTask", () => {
+  let cleanupLoad: () => void;
+  let cleanupExport: () => void;
+
+  beforeAll(() => {
+    cleanupLoad = mockTaskImporterLoadJSZip();
+    cleanupExport = mockTaskConverterExports();
+  });
+
+  afterAll(() => {
+    cleanupLoad();
+    cleanupExport();
+  });
+
+  it("should successfully export a generic notebook task", async () => {
+    const task: GenericNotebookTask = {
+      taskFile: new Blob(["task content"]),
+      data: new Map([["data.txt", new Blob(["data content"])]]),
+      gradingData: new Map([["grading.txt", new Blob(["grading content"])]]),
+      src: new Map([["code.py", new Blob(["print('hello')"])]]),
+      gradingSrc: new Map([["test.py", new Blob(["print('test')"])]]),
+    };
+
+    const result = await TaskConverter.exportExternalCustomTask(task);
+    expect(result).toBeInstanceOf(Blob);
+
+    const resultZip = await loadExportedBlob(result);
+    const regenerated = await resultZip.generateAsync({ type: "blob" });
+
+    expect(regenerated).toBeInstanceOf(Blob);
+    expect(resultZip.file(GenericNotebookFiles.task)).not.toBeNull();
+    expect(
+      resultZip.file(`${GenericNotebookFiles.data}/data.txt`),
+    ).not.toBeNull();
+  });
+});
+
+describe("export and import round-trip", () => {
+  let cleanupLoad: () => void;
+  let cleanupExport: () => void;
+
+  beforeAll(() => {
+    cleanupLoad = mockTaskImporterLoadJSZip();
+    cleanupExport = mockTaskConverterExports();
+  });
+
+  afterAll(() => {
+    cleanupLoad();
+    cleanupExport();
+  });
+
+  it("should be able to import exported CRT internal task", async () => {
+    const task: CrtInternalTask = {
+      taskTemplateFile: new Blob(["template"]),
+      studentTaskFile: new Blob(["student"]),
+      autograderFile: new Blob(["autograder"]),
+      data: new Map([["data.txt", new Blob(["data content"])]]),
+      gradingData: new Map([["grading.txt", new Blob(["grading content"])]]),
+      src: new Map([["code.py", new Blob(["print('hello')"])]]),
+      gradingSrc: new Map([["test.py", new Blob(["print('test')"])]]),
+    };
+
+    const exported = await TaskConverter.exportCrtInternalTask(task);
+    const imported = await TaskConverter.importCrtInternalTask(exported);
+
+    expect(await imported.taskTemplateFile.text()).toBe("template");
+    expect(await imported.data.get("data.txt")?.text()).toBe("data content");
```

Comment:
If you complete the tests with all content checks I am very happy :)

You can also introduce  a function that checks it for all entries and re-use it in multiple tests